flang/test/Lower/io-statement-2.f90

```diff
@@ -106,6 +106,29 @@ subroutine control2() ! I/O condition specifier control flow (use index result)
 write(*,'(8F4.1,I5)',iostat=m) (c,d,j=11,14), j
 end
 
+! CHECK-LABEL: func @_QPloopnest
+subroutine loopnest
+   integer :: aa(3,3)
+   aa = 10
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: BeginExternalListOutput
+   ! CHECK: EnableHandlers
+   ! CHECK: fir.iterate_while
+   ! CHECK: fir.if
+   ! CHECK: fir.iterate_while
+   ! CHECK: fir.if
+   ! CHECK: OutputInteger64
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: fir.result
+   ! CHECK: EndIoStatement
```

Comment:
These CHECK lines seem a bit loose. (For example, it is checking there are 6 fir.result ops, but there could be 14, they could be in a different function, there is no restriction on which nesting level they appear in, etc.)

Reply:
I've added scopes and a little more detail.  The CHECK-LABEL directives should confine verification to the appropriate function.